### PR TITLE
Add Change getter graphQL api

### PR DESCRIPTION
**Pull request by rohithbalaji123**
Part of #612 

### Description
Add `changes` graphQL API to fetch all the changes available to be displayed in admin panel. This is different from `changeLog` in the sense that, `changeLog` also fetches `lastViewedAt` timestamp which is not related to admin panel functionalities. The newly added API internally reuses the same repository methods used for `changeLog`. This also adds the permission check whether the user can view the changes.

### CODE REVIEW

File: backend/app/usecase/changelog/changelog_test.go [hunk 1]
```diff
@@ -448,6 +448,116 @@ func TestPersist_ViewChangeLog(t *testing.T) {
...
+				},
+				{
+					ID:              "54321",
+					Title:           "title 2",
+					SummaryMarkdown: &summaryMarkdown2,
+				},
+			},
+			availableKeys: []keygen.Key{},
+			roles: map[string][]role.Role{
+				"alpha": {role.Admin},
```
Comment (arberiii) [inline backend/app/usecase/changelog/changelog_test.go:494]:
```suggestion
				"alpha": {role.ViewChange},
```
I think is better to test with the user that has the specific role

File: backend/app/adapter/gqlapi/resolver/authquery.go [hunk 2]
```diff
@@ -55,6 +55,26 @@ func (v AuthQuery) ChangeLog() (ChangeLog, error) {
 	return newChangeLog(changeLog, lastViewedAt), err
 }
 
+// Changes retrieves all the changes that exists in the persistent storage
```
Comment (arberiii) [inline backend/app/adapter/gqlapi/resolver/authquery.go:58]:
```suggestion
// Changes retrieves all the changes that exists in the persistent storage.
```
nit: The comments in go should end with a full stop.

**Comment by byliuyang** [comment]
@rohithbalaji123 GraphQL client side query can decide which fields to fetch and ignore those that are not needed.

File: backend/app/adapter/gqlapi/schema.go [hunk 3]
```diff
@@ -47,6 +47,9 @@ type AuthQuery {
 	"""Fetch the all the changes visible to the current user"""
 	changeLog: ChangeLog!
 
+	"""Fetch all the changes that exists in the system"""
+	changes: [Change!]!
```
Comment (byliuyang) [inline backend/app/adapter/gqlapi/schema.go:51]:
```suggestion
	allChanges: [Change!]!
```

File: backend/app/usecase/changelog/changelog.go [hunk 4]
```diff
@@ -27,10 +27,11 @@ func (e ErrUnauthorizedAction) Error() string {
 
 // ChangeLog retrieves change log and create changes.
 type ChangeLog interface {
-	CreateChange(title string, summaryMarkdown *string, user entity.User) (entity.Change, error)
 	GetChangeLog() ([]entity.Change, error)
 	GetLastViewedAt(user entity.User) (*time.Time, error)
 	ViewChangeLog(user entity.User) (time.Time, error)
+	CreateChange(title string, summaryMarkdown *string, user entity.User) (entity.Change, error)
+	GetChanges(user entity.User) ([]entity.Change, error)
```
Comment (byliuyang) [inline backend/app/usecase/changelog/changelog.go:34]:
```suggestion
	GetAllChanges(user entity.User) ([]entity.Change, error)
```

**Comment by rohithbalaji123** [comment]
The reason why I went for a new API is that they are being used for completely different purposes and their fetching mechanism might change in future.
For instance, when we start implementing #595 , that will be specific to user side changelogs, while admin will still need to get all the changes.

**Pull request merged**